kbs/src/plugins/implementations/splitapi/manager.rs

```diff
@@ -0,0 +1,163 @@
+    match SandboxDirectoryMapper::load_from_file(file_path) {
+        Ok(loaded_mapper) => {
+            *mapper = Some(loaded_mapper);
+        }
+        Err(_e) => {
+            // Initialize a new manager
+            *mapper = Some(SandboxDirectoryMapper::new());
+
+            // TODO: check specific errors (file not found or something else) 
+            // and handle those specific errors
+            // bail if there's relevant condition
+        }
+    }
+
+    Ok(())
+}
+
+// Get a reference to the singleton
+fn get_sandbox_directory_mapper() -> Arc<Mutex<Option<SandboxDirectoryMapper>>> {
+    Arc::clone(&SANDBOX_DIRECTORY_MAPPER)
+}
+
+
+#[derive(Debug, Deserialize, Clone, PartialEq)]
+pub struct SplitAPIRepoDesc {
+    #[serde(default)]
+    pub plugin_dir: String,
+}
+
+impl Default for SplitAPIRepoDesc {
+    fn default() -> Self {
+        Self {
+            plugin_dir: DEFAULT_PLUGIN_DIR.into(),
+        }
+    }
+}
+
+
+pub struct CertManager {
+    pub plugin_dir: String,
+    pub mapping_filename: String,
+    mapper: Arc<Mutex<Option<SandboxDirectoryMapper>>>,
+}
+
+
+impl CertManager {
+    pub fn new(repo_desc: &SplitAPIRepoDesc) -> anyhow::Result<Self> {
+        // Create splitapi_res work dir.
+        if !Path::new(&repo_desc.plugin_dir).exists() {
+            fs::create_dir_all(&repo_desc.plugin_dir)?;
+
+            log::info!("Splitapi plugin directory created = {}", repo_desc.plugin_dir);
+        }
+
+        // Initialize directory manager with the content from a file
+        let mapping_file: PathBuf = PathBuf::from(&repo_desc.plugin_dir)
+            .as_path()
+            .join(SANDBOX_DIRECTORY_MAPPING_FILENAME
+        );
+        init_sandbox_directory_mapper(mapping_file.clone())?;
+        log::info!("Directory manager loaded the data from file: {}", mapping_file.display());
+
+        // Initialize the manager
+        Ok(Self {
+            plugin_dir: repo_desc.plugin_dir.clone(),
+            mapping_filename: SANDBOX_DIRECTORY_MAPPING_FILENAME.into(),
+            mapper: get_sandbox_directory_mapper(),
+        })
+    }
+}
+
+#[async_trait::async_trait]
+impl SplitAPIBackend for CertManager {
+    async fn get_server_credential(&self, params: &SandboxParams) -> Result<Vec<u8>> {
+        // Try locking the sandbox directory mapper
+        let mut mapper_guard = self.mapper.lock().map_err(|e| {
+            anyhow!("Failed to lock sandbox directory mapper: {}", e)
+        })?;
+
+        if let Some(mapper) = mapper_guard.as_mut() {
+            let sandbox_dir_info: SandboxDirectoryInfo;
+
+            if let Some(existing_dir) = mapper.get_directory(&params.name) {
+            
+                log::info!("Found existing directory: {:?}", existing_dir.sandbox_dir());
+                sandbox_dir_info = existing_dir.clone();
+
+                //TODO: check if the credentails are already in there
+                // send the existing credentials if they are not expired
```

Comment:
When you create the certs, just store the expiration time alongside them so you can easily see if they are expired.